### PR TITLE
build(all): migrate to spring boot 3

### PR DESCRIPTION
**Pull request by zugazagoitia-eulisa**

### CODE REVIEW

File: domibusConnectorBuilder/domibusConnectorBuilderDeployable/src/main/frontend/index.html
Comment (zugazagoitia-eulisa) [inline domibusConnectorBuilder/domibusConnectorBuilderDeployable/src/main/frontend/index.html:1]: All of the generated files should be ignored: https://vaadin.com/docs/latest/flow/configuration/source-control merge with the gitignore at the root. For reference: https://github.com/eu-LISA/domibusConnector/blob/5e853b8697a0e7f2823aed69956c1f3abca6e8b2/.gitignore

File: domibusConnectorAPI/pom.xml [hunk 1]
```diff
@@ -60,8 +72,8 @@
                 <dependencies>
                     <dependency>
                         <groupId>net.sf.saxon</groupId>
-                        <artifactId>saxon</artifactId>
-                        <version>8.7</version>
+                        <artifactId>Saxon-HE</artifactId>
+                        <version>12.5</version>
```
Comment (zugazagoitia-eulisa) [inline domibusConnectorAPI/pom.xml]: Add a version variable in main pom

File: domibusConnectorController/pom.xml [hunk 2]
```diff
@@ -124,6 +109,25 @@
             <groupId>org.slf4j</groupId>
             <artifactId>jul-to-slf4j</artifactId>
         </dependency>
+        <!-- other dependencies -->
+        <dependency>
+            <groupId>org.projectlombok</groupId>
```
Comment (zugazagoitia-eulisa) [inline domibusConnectorController/pom.xml:114]: Is it needed if we're using the plugin?

File: domibusConnectorController/pom.xml [hunk 3]
```diff
@@ -124,6 +109,25 @@
...
+            <artifactId>artemis-jms-server</artifactId>
+            <exclusions>
+                <exclusion>
+                    <artifactId>jboss-logmanager</artifactId>
+                    <groupId>org.jboss.logmanager</groupId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
```
Comment (zugazagoitia-eulisa) [inline domibusConnectorController/pom.xml]: Can we remove it? Shouldn't be needed with modern Jakarta/Spring annotations

Comment (zugazagoitia-eulisa) [reply domibusConnectorController/pom.xml]: Also it's unsupported, last version in 2015

File: domibusConnectorControllerAPI/pom.xml [hunk 4]
```diff
@@ -37,11 +41,20 @@
...
+            <groupId>jakarta.jms</groupId>
+            <artifactId>jakarta.jms-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>jakarta.platform</groupId>
+            <artifactId>jakarta.jakartaee-api</artifactId>
+        </dependency>
+        <!-- other dependencies -->
         <dependency>
             <groupId>com.google.code.findbugs</groupId>
```
Comment (zugazagoitia-eulisa) [inline domibusConnectorControllerAPI/pom.xml]: Same as the other findbugs

File: domibusConnectorEvidencesToolkit/pom.xml [hunk 5]
```diff
@@ -134,7 +146,7 @@
                     <dependency>
                         <groupId>org.glassfish.jaxb</groupId>
                         <artifactId>jaxb-runtime</artifactId>
-                        <version>2.3.5</version>
+                        <version>4.0.5</version>
```
Comment (zugazagoitia-eulisa) [inline domibusConnectorEvidencesToolkit/pom.xml]: Extract version to parent pom

**Review by zugazagoitia-eulisa — COMMENTED** [review]
Other that the current comments, replace` javax.annotation.Nonnull` with `jakarta.annotation.Nonnull` projectwide (a replace is enough). Also for other `javax.annotation` classes.

File: domibusConnectorController/pom.xml (hunk 3 above)
Comment (komi-eulisa) [reply domibusConnectorController/pom.xml]: Removed

File: domibusConnectorControllerAPI/pom.xml (hunk 4 above)
Comment (komi-eulisa) [reply domibusConnectorControllerAPI/pom.xml]: Removed

File: domibusConnectorAPI/pom.xml (hunk 1 above)
Comment (komi-eulisa) [reply domibusConnectorAPI/pom.xml]: Done

File: domibusConnectorController/pom.xml (hunk 2 above)
Comment (komi-eulisa) [reply domibusConnectorController/pom.xml:114]: Which plugin ?

File: domibusConnectorEvidencesToolkit/pom.xml (hunk 5 above)
Comment (komi-eulisa) [reply domibusConnectorEvidencesToolkit/pom.xml]: Done

**Comment by komi-eulisa** [comment]
> Other that the current comments, replace` javax.annotation.Nonnull` with `jakarta.annotation.Nonnull` projectwide (a replace is enough). Also for other `javax.annotation` classes.

Done

File: domibusConnectorBuilder/domibusConnectorBuilderDeployable/src/main/frontend/index.html
Comment (komi-eulisa) [reply domibusConnectorBuilder/domibusConnectorBuilderDeployable/src/main/frontend/index.html:1]: Done

File: domibusConnectorController/pom.xml (hunk 2 above)
Comment (zugazagoitia-eulisa) [reply domibusConnectorController/pom.xml:114]: My bad, that's gradle.

**Pull request merged**